pkg/chunk/cassandra/storage_client.go

```diff
@@ -83,6 +79,14 @@ func (cfg *Config) session() (*gocql.Session, error) {
 	}
 	cfg.setClusterConfig(cluster)
 
+	session, err := cluster.CreateSession()
+	if err == nil || err != gocql.ErrNoConnectionsStarted {
+		return session, nil
```

Comment:
We enter this `if` even in case of error, but we never return `err`.  Please rework this logic to make it clearer. A clearer approach may be:

```
if err == nil {
  return session, nil
}

// TODO: explain what ErrNoConnectionsStarted is
if err != gocql.ErrNoConnectionsStarted {
  return nil, errors.WithStack(err)
}
```